src/plugins/plugin.legend.js

```diff
@@ -42,13 +42,14 @@ defaults._set('global', {
 			// lineJoin :
 			// lineWidth :
 			// legendSymbol : which symbol to use for the legend start with pointStyle then dataset level and finally options/legend/label level
-			// boxWidth : if not usePointStyle and legendSymbol ends with Lg then use boxWidth value otherwise use fontSize * Math.SQRT2
+			// legendSymbolLarge: boolean impacting boxWidth
+			// boxWidth : if LegendSymbolLarge is true used boxWidth otherwise use fontSize * Math.SQRT2
 			generateLabels: function(chart) {
 				var data = chart.data;
 				var opts = chart.options.legend.labels;
 				var fontSize = helpers.valueOrDefault(opts.fontSize, defaults.global.defaultFontSize);
 				return helpers.isArray(data.datasets) ? data.datasets.map(function(dataset, i) {
-					var itemLegendSymbol = (opts.usePointStyle && (dataset.pointStyle || defaults.global.elements.point.pointStyle)) || dataset.legendSymbol || (opts.legendSymbol);
+					var itemLegendSymbolLarge = typeof dataset.legendSymbolLarge !== 'undefined' ? dataset.legendSymbolLarge : (typeof opts.legendSymbolLarge !== 'undefined' ? opts.legendSymbolLarge : (opts.usePointStyle ? false : true));
```

Comment:
You might be able to use `helpers.getValueOrDefault` since this pattern is used in a few places.